src/odemis/gui/cont/acquisition/fastem_acq.py

```diff
@@ -1197,6 +1205,93 @@ def __init__(
         # enable/disable calibration buttons if calibrating
         self._tab_data_model.is_calibrating.subscribe(self._on_is_calibrating)
 
+    def _on_visibility_btn(self, evt):
+        """Toggle the visibility of the calibration region based on the button state."""
+        current_sample = self._main_data_model.current_sample.value
+        focussed_view = self._main_tab_data.focussedView.value
+        if not (current_sample and focussed_view):
+            return
+
+        scintillator_num = int(focussed_view.name.value)
+
+        # Determine which calibration button was pressed
+        btn = evt.GetEventObject()
+        if btn == self._calib_1_vis_btn:
+            calibration_key = CALIBRATION_1
+        elif btn == self._calib_2_vis_btn:
+            calibration_key = CALIBRATION_2
+        elif btn == self._calib_3_vis_btn:
+            calibration_key = CALIBRATION_3
+
+        calibration = current_sample.scintillators[scintillator_num].calibrations[calibration_key]
+
+        # Toggle visibility based on button state
+        is_visible = btn.GetToggle()
+        calibration.shape.active.value = is_visible
+        if is_visible:
+            calibration.shape.cnvs.add_world_overlay(calibration.shape)
+        else:
+            calibration.shape.cnvs.remove_world_overlay(calibration.shape)
+
+        calibration.shape.cnvs.request_drawing_update()
+
+    def add_calibration_control(self, label_text, value=True, pos_col=1, span=wx.DefaultSpan):
+        """ Add a calibration control to the calibration settings panel
+
+        :param label_text: (str) Label text to display
+        :param value: (bool) Value to display (True == checked)
+        :param pos_col: (int) The column index in the grid layout where the checkbox will be placed.
+                        For example:
+                        - `pos_col=0` positions the checkbox in the first column.
+                        - `pos_col=1` positions it in the second column.
+        :param span: (tuple) the row and column spanning attributes of items in a GridBagSizer.
+        :returns: (wx.StaticText, ImageToggleButton, wx.CheckBox)
+            A tuple containing the label, visibility button and checkbox
+        """
+        self.calibration_panel.clear_default_message()
+        self.calibration_panel.Layout()
+        self.calibration_panel.num_rows += 1
+
+        lbl_ctrl, visibility_btn = self._add_side_label_with_toggle(label_text)
+        value_ctrl = wx.CheckBox(self.calibration_panel, wx.ID_ANY, style=wx.ALIGN_RIGHT | wx.NO_BORDER)
+        self.calibration_panel.gb_sizer.Add(value_ctrl, (self.calibration_panel.num_rows, pos_col), span=span,
+                                            flag=wx.EXPAND | wx.TOP | wx.BOTTOM, border=5)
+        value_ctrl.SetValue(value)
+
+        return lbl_ctrl, visibility_btn, value_ctrl
+
+    def _add_side_label_with_toggle(self, label_text):
```

Comment:
The function name seems like a general utility function, although the return value is specific. Maybe rename the function or the function contents  to make it either general or specific for clarity purpose.

Reply:
good point, I'll rename it to `_add_calibration_label_with_toggle`